kod/object/active/holder/room/monsroom/bossroom/ratdungeon.kod

```diff
@@ -0,0 +1,251 @@
+      plHenchmenTemplate = [ [&GiantRatSoldier, 33, 40],
+                             [&GiantRatSoldier, 32, 40],
+                             [&GiantRatSoldier, 29, 37],
+                             [&GiantRatSoldier, 27, 39],
+                             [&GiantRatSoldier, 29, 41],
+                             [&GiantRatSoldier, 35, 37],
+                             [&GiantRatSoldier, 37, 39],
+                             [&GiantRatSoldier, 37, 40] ];
+
+      plBossTreasureTemplate = [ [&Gauntlet, 33, 40, 0, 0],
+                                 [&FineLute, 29, 37, 0, 0],
+                                 [&ScaleArmor, 27, 39, 0, 0],
+                                 [&Axe, 35, 37, 0, 0] ,
+                                 [&HealWand, 36, 41, 0, 0]
+                               ];
+
+
+      return;
+   }
+
+   CreateStandardObjects()
+   {
+      local lSkulls;
+
+      lSkulls = [ [13, 13, 15, 36, 30, 26, 32],
+                  [36, 18, 3, 8, 42, 37, 36] ];
+
+      Send(self,@CreateObjFromList,#lObjlist=lSkulls,
+            #oObjectClassName=&Skull);
+            
+      return;
+   }
+
+
+   CreateStandardExits()
+   {
+      propagate;
+   }
+
+%%%%%%%%%%%%%%%%%%%%%
+% Monster death code%
+%%%%%%%%%%%%%%%%%%%%%
+%
+
+   BossKilledTrigger()
+   {
+      local i, each_obj;
+
+      Send(self,@SomethingWaveRoom,#wave_rsc=RatDungeon_door_up_sound);
+      Send(self,@SetSector,#sector=1,
+            #animation=ANIMATE_FLOOR_LIFT,
+            #height=770,#speed=20);
+      Send(self,@SetSector,#sector=2,
+            #animation=ANIMATE_FLOOR_LIFT,
+            #height=770,#speed=20);
+      Send(self,@SetSector,#sector=3,
+            #animation=ANIMATE_FLOOR_LIFT,
+            #height=770,#speed=20);
+      
+      pbGenerateMonsters = FALSE;
+      
+      foreach i in plActive
+      {
+         each_obj = Send(self,@HolderExtractObject,#data=i);
+         if IsClass(each_obj,&GiantRat)
+            OR IsClass(each_obj,&GiantRatSoldier)
+         {
+            Send(each_obj,@Delete);
+         }
+         
+         if IsClass(each_obj,&Player)
+         {
+            Send(each_obj,@MsgSendUser,#message_rsc=RatKing_Slain_msg);
+         }
+      }
+            
+      propagate;
+   }
+
+%
+%%%%%%%%%%%%%%%%%%%%%%%%%
+% end Monster death code%
+%%%%%%%%%%%%%%%%%%%%%%%%%
+
+
+%%%%%%%%%%%%%%%%%%%%%%%
+% Boss resetting code %
+%%%%%%%%%%%%%%%%%%%%%%%
+%
+   ResetBoss()
+   {
+      Send(self,@SetSector,#sector=1,
+            #animation=ANIMATE_FLOOR_LIFT,
+            #height=670,#speed=0);
+      Send(self,@SetSector,#sector=2,
+            #animation=ANIMATE_FLOOR_LIFT,
+            #height=570,#speed=0);
+      Send(self,@SetSector,#sector=3,
+            #animation=ANIMATE_FLOOR_LIFT,
+            #height=100,#speed=0);
+      
+      pbGenerateMonsters = TRUE;
+
+      propagate;
+   }
+
+%
+%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% End Boss resetting code %
+%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+   ReqLeaveHold(what = $)
```

Comment:
This message has an error (it should propagate if object not in the boss treasure list) however the exact message you need is already in bossroom.kod so you can just delete this ReqLeaveHold entirely.